### PR TITLE
[MM-28408]Add e2e to test disabling textbox in archived channel MM-T1716

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Add e2e tests for check if textboxes are disabled for archived channels
Add a utility for archiving channels
Add a folder for organizing archive channel tests

PS: Ticket MM-27211 has multiple TM4J tests but this PR is for the first one alone MM-T1716
#### Ticket Link
Story: https://mattermost.atlassian.net/browse/MM-27211
Subtask: https://mattermost.atlassian.net/browse/MM-28408

### CODE REVIEW

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel
+
+import {testWithConfig} from '../../../support/hooks';
```
Comment (srkgupta) [inline e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:13]: Correct path:
`import {testWithConfig} from '../../support/hooks';`

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('MM-T1716 Archived channels are read only - Cannot post', () => {
```
Comment (srkgupta) [inline e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:15]: Suggested change:

```
describe('Archived channels', () => {
```

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    });
+
+    before(() => {
+        // # Login as test user and visit create channel
+        cy.apiInitSetup({loginAfter: true}).then(({team, channel}) => {
+            cy.visit(`/${team.name}/channels/${channel.name}`);
+        });
+    });
+
+    it('Text box in center channel and in RHS should not be visible', () => {
```
Comment (srkgupta) [inline e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:29]: Suggested change:
```
it('MM-T1716 Archived channels are read only - Cannot post', () => {
```

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
```
Comment (srkgupta) [inline e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:10]: New tests dont have the prod tag. Please remove this line. Once the test works fine for 5 days, the QA team will add this line again.

**Review by srkgupta — CHANGES_REQUESTED** [review]
Thanks @sudheerDev. Overall the test looks great. Suggested some minor changes.

**Review by Willyfrog — APPROVED** [review]

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js (hunk 1 above)
Comment (sudheerDev) [reply e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:13]: Based on the folder structure this is actually right from this folder. I am not a fan of relative URLs when it exceeds more than `../` but as I mostly saw the usage with the relative path I went with it.

**Review by jgilliam17 — APPROVED** [review]
Thank you @sudheerDev 
Looks good to merge.
- Test case, steps, details and links updated.

File: e2e/cypress/support/ui_commands.js [hunk 5]
```diff
@@ -432,3 +432,10 @@ Cypress.Commands.add('checkRunLDAPSync', () => {
         }
     });
 });
+
+Cypress.Commands.add('archiveCurrentChannel', () => {
```
Comment (Willyfrog) [inline e2e/cypress/support/ui_commands.js:436]: oh! I think we both have a ui command for archive current channel, we might need to remove one at some point (but they are almost the same, [different placement though](https://github.com/mattermost/mattermost-webapp/pull/6358/files#diff-08c1f6e408556f0b68412a28840bda2aR46))
/cc @saturninoabril

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js (hunk 1 above)
Comment (srkgupta) [reply e2e/cypress/integration/archived_channel/archive_channel_post_spec.js:12]: @sudheerDev the Cypress test does not runs as the path is not correct. Can you please fix the correct path to `support/hooks`. It should be `../../support/hooks`. Only then the test works fine. Else we see this error. 

Screenshot 2020-09-11 at 4 50 43 PM

**Review by srkgupta — CHANGES_REQUESTED** [review]
Please see my above comments regarding the import path. It's causing test to error

File: e2e/cypress/support/ui_commands.js (hunk 5 above)
Comment (saturninoabril) [reply e2e/cypress/support/ui_commands.js:436]: Yeah, I think the one you have is what we'll use moving forward. So if it gets merged first, then I'd suggest to update and use it in this PR 😄

Comment (sudheerDev) [reply e2e/cypress/support/ui_commands.js]: Changed it to use the other method

File: e2e/cypress/integration/archived_channel/archive_channel_post_spec.js (hunk 1 above)
Comment (sudheerDev) [reply e2e/cypress/integration/archived_channel/archive_channel_post_spec.js]: Wierd it was working for me earlier.

**Review by srkgupta — APPROVED** [review]
Thanks @sudheerDev. Tested and passed. Changes LGTM

**Pull request merged**